test/gtest/uct/uct_test.cc

```diff
@@ -1213,6 +1213,15 @@ void uct_test::entity::flush() const {
     ASSERT_UCS_OK(status);
 }
 
+void uct_test::entity::flush(const uct_test &test) const {
+    ucs_status_t status;
+    do {
+        test.progress();
+        status = uct_iface_flush(m_iface, 0, NULL);
```

Comment:
it is a potential deadlock. Also we already have flush func which suits your needs on line 586

Reply:
> Also we already have flush func which suits your needs on line 586

I know it, it flushes all entities. The goal of this function is to flush a specific iface, but progress all available entities

> it is a potential deadlock

why?

Reply:
what is the problem to flush all entities?
typically we use deadline to not hang in the infinite loop in case of error
We have 15 mins general timeout, but 10-20 sec would be enough for the flush (like it is done in the existing test)

Reply:
> what is the problem to flush all entities?

not a problem, can use it

> typically we use deadline to not hang in the infinite loop in case of error
> We have 15 mins general timeout, but 10-20 sec would be enough for the flush (like it is done in the existing test)

ok, can we use `uct_test::flush()`

Reply:
done